fpm/measurement.php

```diff
@@ -21,7 +21,8 @@
 
 namespace Google\FirstPartyLibrary;
 
-/* Start of Site Kit modified code. */
+/* Start of Site Kit modified code.
+ * This block is an addition to the original file. */
 if ( isset( $_GET['healthCheck'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
Why do we need this block? I think we can remove it completely without loosing anything because the original script at the bottom of this file will send a healthcheck request to the Google server and return the same `ok` if everything is okay, right?

Alternatively, if the idea for this block is to verify whether we can make direct requests to php files located in the `wp-content/plugins` folder, then we can create one more file in the `fpm` folder that will just echo `ok`:

```php
<?php
// fpm/scriptcheck.php

echo 'ok';
exit;
```

In this case, the `mesurement.php` file will have no our changes and it will be super easy to update when the new version is released by the Google team.

Reply:
This was discussed to some extent on [Slack](https://fueled10up.slack.com/archives/C077DB69US2/p1732086113298819).

Taking another look, I can see that at present it's not possible to make a call to the `/healthy` endpoint on the Google server via the `measurement.php` script without modifying its logic.

Trying to use pass the `/healthy` path through using the `s` query parameter (bearing in mind the `id` param is also [required](https://github.com/google/site-kit-wp/blob/62d3e2aa9f026af081b1981cee4c9bd46ced2223/fpm/measurement.php#L66-L69)), e.g.:

`http://sitekit.10uplabs.com/wp-content/plugins/google-site-kit/fpm/measurement.php?id=GT-12345&s=/healthy`

...results in a request to a URL like this, which returns a `404`: `https://GT-12345.fps.goog/PHP_FPM_REPLACE_PATH/healthy&uip=10.0.0.1`.

We _could_ provide an additional file e.g. `scriptcheck.php` as you've suggested, however I would prefer to keep the current approach for now, as although it involved this small modification to `measurement.php` it provides a more robust check that verifies the script itself is accessible, whereas `scriptcheck.php` would only verify _that_ script is accessible, which could be a false positive in some scenarios.

The Google team are considering adding support for the health check to the FPM library, if/when that lands we can update Site Kit's health check to use it, but in the meantime I'd suggest we keep the current approach.

Even if we do want to change it, it's out of scope for the current issue and something we should address post launch rather than adding another launch blocking issue.